src/test/ui/token/issue-10636-2.stderr

```diff
@@ -1,25 +1,29 @@
 error: incorrect close delimiter: `}`
-  --> $DIR/issue-10636-2.rs:18:1
+  --> $DIR/issue-10636-2.rs:19:1
    |
 LL | pub fn trace_option(option: Option<isize>) {
    |                                            - close delimiter possibly meant for this
 LL |     option.map(|some| 42;
    |               - un-closed delimiter
 ...
-LL | } //~ ERROR: incorrect close delimiter
+LL | }
    | ^ incorrect close delimiter
 
 error: expected one of `)`, `,`, `.`, `?`, or an operator, found `;`
   --> $DIR/issue-10636-2.rs:15:25
    |
 LL |     option.map(|some| 42;
-   |                         ^ expected one of `)`, `,`, `.`, `?`, or an operator here
+   |               -         -
```

Comment:
TODO: special case that when there're zero width and a 1-width span labels overlapping, to pick the more primary over secondary, regardless of width (the second span above should be `^` instead of `-`).